src/inference/include/openvino/runtime/intel_gpu/properties.hpp

```diff
@@ -115,6 +115,13 @@ static constexpr Property<ov::hint::Priority> host_task_priority{"GPU_HOST_TASK_
  * @ingroup ov_runtime_ocl_gpu_prop_cpp_api
  */
 static constexpr Property<int64_t> available_device_mem{"AVAILABLE_DEVICE_MEM_SIZE"};
+
+/**
+ * @brief This key enables GPU plugin to optimize memory usage on dynamic shape execution.
+ * Note that setting this option true can cause peformance drop.
+ * @ingroup ov_runtime_ocl_gpu_prop_cpp_api
+ */
+static constexpr Property<bool> aggressive_memory_opt{"AGGRESSIVE_MEMORY_OPT"};
```

Comment:
Shouldn't we add corresponding Python API property as well?

Reply:
I tested in python app and it worked. Should we do something else?

Reply:
It provides more convenient python namespace for hint, something like intel_gpu_hint.aggressive_memory_opt, but if functionally it works, I think we may add this later

Reply:
I see! Lets add it in the next PR. Thanks!